src/main/java/io/redisearch/Client.java

```diff
@@ -0,0 +1,69 @@
+package io.redisearch;
+
+import io.redisearch.aggregation.AggregationRequest;
+import io.redisearch.client.AddOptions;
+import io.redisearch.client.SuggestionOptions;
+
+import java.util.List;
+import java.util.Map;
+
+public interface Client {
+
+    String INCREMENT_FLAG = "INCR";
+    String PAYLOAD_FLAG = "PAYLOAD";
+    String WITHPAYLOADS_FLAG = "WITHPAYLOADS";
+    String MAX_FLAG = "MAX";
+    String FUZZY_FLAG = "FUZZY";
+    String WITHSCORES_FLAG = "WITHSCORES";
+
+    boolean createIndex(Schema schema, io.redisearch.client.Client.IndexOptions options);
+
+    SearchResult search(Query q);
+
+    AggregationResult aggregate(AggregationRequest q);
+
+    String explain(Query q);
+
+    boolean addDocument(String docId, double score, Map<String, Object> fields, boolean noSave, boolean replace, byte[] payload);
+
+    boolean addDocument(Document doc, AddOptions options);
```

Comment:
This (`addDocument(Document, AddOptions)`) should be the canonical interface. The other helper methods are there for legacy purposes.